Core/clim-basic/extended-streams/stream-input.lisp

```diff
@@ -43,6 +44,104 @@
               (eql modifiers +shift-key+))
       (keyboard-event-character event))))
 
+
+(deftype input-gesture-object ()
+  '(or character event))
+
+;;; CLIM's input kernel is only hinted in the specification. -- jd 2019-06-24
+(defclass input-stream-kernel (standard-sheet-input-mixin)
+  ((input-buffer :initarg :input-buffer :accessor stream-input-buffer))
+  (:default-initargs
+   :input-buffer (or *input-buffer* (make-instance 'concurrent-event-queue))))
```

Comment:
indeed it is a suitable implementation. the fallback is here to have a working input stream even when *input-buffer* is not bound. I'd rather avoid renaming the event queue classes. I could add some lengthy comment in the source code explaining this. Notice, that the internal representation of the input-buffer is exposed only in implementation of the input-stream-kernel (and, superficially for performance, in stream-read-line defined on standard-input-stream), so it doesn't "leak" all over the place. Another point worth mentioning is that previously the input-buffer and event-queue were treated as interchangeable names, so from the "confusion" point of view what we do now is at least less unclear.

Reply:
I think some explanation is appropriate since the commit message emphasizes the separation. I'm not concerned about leaking anything to application programmers but about maintenance programmers looking at this commit.

Reply:
added comment